tsdb/querier.go

```diff
@@ -106,14 +109,29 @@ func (q *querier) lvals(qs []Querier, n string) ([]string, error) {
 }
 
 func (q *querier) Select(ms ...*labels.Matcher) (SeriesSet, error) {
+	if len(q.blocks) != 1 {
+		return q.SelectSorted(ms...)
+	}
+	// Sorting Head series is slow, and unneeded when only the
+	// Head is being queried. Sorting blocks is a noop.
+	return q.blocks[0].Select(ms...)
+}
+
+func (q *querier) SelectSorted(ms ...*labels.Matcher) (SeriesSet, error) {
```

Comment:
do we need this publicly?

Reply:
It's the Query interface that we're using everywhere, having shadow internal interfaces that are slightly different tends to end up being hard to maintain

Reply:
Yes please (: